wake/actions/cart/updateItemQuantity.ts

```diff
@@ -65,18 +46,35 @@ const action = async (
     throw new HttpError(400, "Missing cart cookie");
   }
 
-  let data = await removeFromCart(props, cartId, ctx, headers);
+  /* 
+  * get cart
+  * find the current product on cart
+  * the current amount
+  * calculate the difference between the current item amount and requested new amount 
+  */
 
-  if (props.quantity > 0) {
-    data = await addToCart(props, cartId, ctx, headers);
+  const cart = await ctx.invoke.wake.loaders.cart(props, req)
+  const item = cart.products?.find(item => item?.productVariantId === props.productVariantId)
+  const quantityItem = item?.quantity ?? 0
+  const quantity = props.quantity - quantityItem
+
+  let checkout
+
+  if (props.quantity > 0 && quantity > 0) {
+    checkout = await ctx.invoke.wake.actions.cart.addItem({...props, quantity})
+  } else {
+    const data = await removeFromCart({...props, quantity:-quantity}, cartId, ctx, headers);
```

Comment:
based on the formatting, I think you did miss a `deno task check` can you run it before get it merged?